### PR TITLE
BR-58 Convert usng4j to Pipeline Job

### PR DESCRIPTION
**Pull request by TonyMorrison**
#### What does this PR do?
Takes the codice/usng4j job and converts it to Jenkins pipeline

#### Who is reviewing it? 
@shaundmorris 
@LinkMJB  

#### Select relevant component teams: 
@codice/security

#### Ask 2 committers to review/merge the PR and tag them here.
@shaundmorris 
@bdeining 

#### How should this be tested?
Three main stages should build after checking out the code
1. Full Build no ITest
2. Full Build
3. Deploy (may not Build for PR build)

Full test result from my repo
http://jenkins.phx.connexta.com/service/jenkins/job/W.I.P./job/usng4j_TM/job/master/

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        // Checkout the repository
+        stage('Checkout repo') {
+            steps {
+                retry(3) {
+                    checkout([$class: 'GitSCM', branches: [[name: "refs/heads/${BRANCH_NAME}"]], doGenerateSubmoduleConfigurations: false, extensions: [[$class: 'WipeWorkspace'], [$class: 'LocalBranch', localBranch: "${BRANCH_NAME}"], [$class: 'CloneOption', timeout: 30, shallow: true]], submoduleCfg: [], userRemoteConfigs: scm.userRemoteConfigs ])
+                }
+            }
+        }
+
+        stage('Full Build except itest') {
```
Comment (shaundmorris) [inline Jenkinsfile]: I'm not sure we need this stage at all in this repo. There are no itests.

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+            when {
+                allOf {
+                    expression { env.CHANGE_ID == null }
+                    expression { env.BRANCH_NAME ==~ /((?:\d*\.)?\d*\.x|master)/ }
+                    environment name: 'JENKINS_ENV', value: 'prod'
+                }
+            }
+            steps{
+                withMaven(maven: 'maven-latest', jdk: 'jdk8-latest', globalMavenSettingsConfig: '51e52749-c47a-4e11-9c58-0adf485626f5', mavenSettingsConfig: 'codice-maven-settings', mavenOpts: '${LINUX_MVN_RANDOM}') {
+                    sh 'mvn deploy -nsu -DskipTests=true -Dpmd.skip=true -Dfindbugs.skip=true -Dcheckstyle.skip=true -Djacoco.skip=true'
```
Comment (shaundmorris) [inline Jenkinsfile]: You can remove some of these flags: documentation (profile doesn't exist), project doesn't use pmd, findbugs, checkstyle.

**Review by shaundmorris — APPROVED** [review]

**Review by LinkMJB — APPROVED** [review]

File: Jenkinsfile (hunk 1 above)
Comment (TonyMorrison) [reply Jenkinsfile]: Did I categorize this stage correctly, based on the flags set? I can easily remove this stage.

Comment (TonyMorrison) [reply Jenkinsfile]: Removed stage

File: Jenkinsfile (hunk 2 above)
Comment (TonyMorrison) [reply Jenkinsfile]: Removed flags

**Pull request merged**